pkg/kubectl/apply/element.go

```diff
@@ -419,3 +431,8 @@ func (e HasElementData) HasLocal() bool {
 func (e HasElementData) HasRemote() bool {
 	return e.remoteSet
 }
+
+// ConflictDetector defines the capability to detect conflict. An element can examine remote/recorded value to detect conflict.
+type ConflictDetector interface {
+	DetectConflict() error
```

Comment:
```go
HasConflict() error
```

Reply:
I think normally **Has...** method returns a boolean, while DetectConflict() which is a executive behavior returns error. Dose it make sense?